File: simulators/de.unistuttgart.iste.sqa.mpw.karasimulator.java/examples/src/main/java/de/unistuttgart/kara/examples/Example01.java
package de.unistuttgart.kara.examples;

import de.unistuttgart.kara.main.SimpleKaraGame;

public class Example01 extends SimpleKaraGame {
    public static void main(final String[] args) {
        createInstance(Example01.class);
    }

    @Override
    protected void run() {
        displayInNewGameWindow();
        kara.move();
        kara.turnRight();
        kara.move();
        kara.removeLeaf();
        kara.turnLeft();
        kara.turnLeft();
        kara.move();
        kara.putLeaf();
        kara.turnLeft();
        kara.move();
        kara.turnRight();
        kara.turnRight();
    }

}

File: simulators/de.unistuttgart.iste.sqa.mpw.karasimulator.cpp/examples/Example01.cpp
#include "SimpleKaraGame.h"

using namespace kara;

class Example01 : public karasimulator::SimpleKaraGame {
protected:
    void run() override;
};

void Example01::run() {
    kara.move();
    kara.turnRight();
    kara.move();
    kara.removeLeaf();
    kara.turnLeft();
    kara.turnLeft();
    kara.move();
    kara.putLeaf();
    kara.turnLeft();
    kara.move();
    kara.turnRight();
    kara.turnRight();
}

int main() {
    karasimulator::SimpleKaraGame::createInstance<Example01>();
    return 0;
}

